Handle malformed requests

argerus opened the pull request:
Signed-off-by: John Argérus <john.argerus@se.bosch.com>

SebastianSchildt commented:
Doesn't work for me. I created a token and tried with testclient. No matter I use wildcard permission or allow access to specific nodes in the tree, I always get 403 doing a get request. 

For example my token was

```
{
  "sub": "1234567890",
  "name": "John Doe",
  "admin": true,
  "iat": 1516239022,
  "exp": 1616239022,
  "w3c-viss": {
  "Vehicle.Acceleration.Vertical": "rw"
  }
}
```

(I can also see the server parses and prints the rules correctly when I give that token via testclient), but I get
```
Enter Token 
eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6MTUxNjIzOTAyMiwiZXhwIjoxNjE2MjM5MDIyLCJ3M2MtdmlzcyI6eyJWZWhpY2xlLkFjY2VsZXJhdGlvbi5WZXJ0aWNhbCI6InJ3In19.HqKsCQjtMbOuMKXoh0vrHQCjhLmEQp1Mfp35_FWE4DCIKumlBvN29jJcT2e_2GzuI8T4333zpvaQtmp-Tuq_qkMYno80L3tPzR27AQuUdbYScx0JNlCsYyR70-BSCnBgNbD5HVsPqmmFAwSQrztaW0DPAv9LaWWBFLCBqhujAW3uCtz2b-l6fjNBf29rLUSGwUqJlQectLc0iK-VT1tWP7MswjiKBdVTpALMdqNUkcwftsGSMYXK2-Q-vR63-sMpRh0218H_2oPDsMTKnKUWk7XPud-s6ggVa--aNsReM0w0tLKXcF6947hcZR8cdefIx55Tuc0SmK5jXBTPl9uQwQ
Response >> {
    "TTL": 1616239022,
    "action": "authorize",
    "requestId": 1238,
    "timestamp": 1568811971
}
Enter vss path eg : Signal.Drivetrain.InternalCombustionEngine.RPM 
Vehicle.Acceleration.Vertical
Enter vis Function eg: authorize, get, set, getmetadata 
get
Response >> {
    "action": "get",
    "error": {
        "message": "No read access to Vehicle.Acceleration.Vertical",
        "number": 403,
        "reason": "Forbidden"
    },
    "requestId": 1234,
    "timestamp": 1568811985
}
```

maybe I did not configure it correctly, so I tried unit-tests, but they are not all that happy either

```
*** 12 failures are detected (7 failures are expected) in the test module "w3c-unit-test"
```

So I guess something is seriously broken here?

argerus commented:
This had me really confused. Util I realized that the claim in the token should be `w3c-vss` (and _not_ `w3c-viss`).

Using that, things work from what I can tell. Haven't analysed why the unit tests fails yet though.

argerus commented:
Alright, 4 of the unit test fail because I changed an error message in this commit:
```
msg << "No read access to  " << getReadablePath(jPath);
                          ^
   I removed this extra space
```
and the unit tests expected that to be there...

Still 8 failures (7 expected, whatever that means)

SebastianSchildt commented:
With the correct token it works. **Can be merged**  (I still think the unit test is not very... elegant... but at least no regression, but rather future work :) )
